chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -1,53 +1,87 @@
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.HashMap;
-import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.gk.model.GKInstance;
 import org.gk.model.InstanceDisplayNameGenerator;
-import org.gk.model.PersistenceAdaptor;
 import org.gk.model.ReactomeJavaConstants;
 import org.gk.persistence.MySQLAdaptor;
 import org.gk.persistence.MySQLAdaptor.AttributeQueryRequest;
 import org.gk.schema.InvalidAttributeException;
 import org.gk.schema.InvalidAttributeValueException;
-import org.gk.schema.SchemaClass;
-import org.gk.util.GKApplicationUtilities;
+import org.reactome.release.common.database.InstanceEditUtils;
 
 import uk.ac.ebi.chebi.webapps.chebiWS.client.ChebiWebServiceClient;
 import uk.ac.ebi.chebi.webapps.chebiWS.model.ChebiWebServiceFault_Exception;
 import uk.ac.ebi.chebi.webapps.chebiWS.model.DataItem;
 import uk.ac.ebi.chebi.webapps.chebiWS.model.Entity;
 
-
-public class ChebiUpdater {
+public class ChebiUpdater
+{
 	private static final Logger logger = LogManager.getLogger();
+	private static final Logger refMolNameChangeLog = LogManager.getLogger("molNameChangeLog");
+	private static final Logger refMolIdentChangeLog = LogManager.getLogger("molIdentChangeLog");
+	private static final Logger refEntChangeLog = LogManager.getLogger("refEntChangeLog");
+	private static final Logger duplicatesLog = LogManager.getLogger("duplicatesLog");
+	private static final Logger failedChebiLookupsLog = LogManager.getLogger("failedChebiLookupsLog");
 	private boolean testMode = true;
 	private MySQLAdaptor adaptor;
 	private ChebiWebServiceClient chebiClient = new ChebiWebServiceClient();
-	private StringBuilder identifierSB = new StringBuilder();
 	private StringBuilder formulaUpdateSB = new StringBuilder();
 	private StringBuilder formulaFillSB = new StringBuilder();
-	private StringBuilder nameSB = new StringBuilder();
 	private StringBuilder duplicatesSB = new StringBuilder();
-
+	private Map<GKInstance, List<String>> referenceEntityChanges = new HashMap<GKInstance, List<String>>();
 	private long personID;
+	private boolean useCache;
+	// A Comparator object that will compare GKInstances, assuming that they are of the "Person" type, with a surname and firstname.
+	private Comparator<GKInstance> personComparator = new Comparator<GKInstance>()
+		{
+			@Override
+			public int compare(GKInstance o1, GKInstance o2)
+			{
+				try
+				{
+					int surnameCompare = ((String)o1.getAttributeValue(ReactomeJavaConstants.surname)).compareTo((String)o2.getAttributeValue(ReactomeJavaConstants.surname));
+					if (surnameCompare == 0)
+					{
+						return ((String)o1.getAttributeValue(ReactomeJavaConstants.firstname)).compareTo((String)o2.getAttributeValue(ReactomeJavaConstants.firstname));
```

Comment:
Extracting to a private method shared for lines 63 and 66 might make this clearer

Reply:
Yeah, there are a few ways this could be made more readable, I guess. I'll look into it.